### DISCUSSION TITLE
Specify Grafana backend MySQL table engine to be used

### DISCUSSION THREAD

**Discussion by mmotoszko** [grafana/grafana#65529] [Platform]
**What would you like to be added:**
A config entry enforcing sql engine when using MySQL as Grafana backed DB.  If it's empty it can default to the currently used one, but if it's not empty then it's value is used, even if it's not a valid sql engine.

From what I understand sql engine is hardcoded in https://github.com/grafana/grafana/blob/main/pkg/services/sqlstore/migrator/dialect.go line 165 as "ENGINE=InnoDB".

**Why is this needed:**
In my use case, I need to have "ENGINE=ndbcluster" to allow MySQL Cluster replication on those tables. 

Currently, instead of just starting my Grafana container, what I need to when creating Grafana DB from scratch, is the following: 
- start my Grafana container to populate 'grafana' DB
- stop my Grafana container
- mysqldump grafana DB
- run `sed -i 's/ENGINE=InnoDB/ENGINE=ndbcluster/g' ./grafana_full_dump.sql`

- import altered Grafana DB dump
- start my Grafana container

This is a time consuming and error-prone process, especially when running multiple Grafana containers on different environments.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82771. Feel free to continue the discussion around this there. Thank you!
